internal/cmd/opensearch/instance/create/create.go

```diff
@@ -0,0 +1,253 @@
+package create
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"stackit/internal/pkg/args"
+	"stackit/internal/pkg/confirm"
+	cliErr "stackit/internal/pkg/errors"
+	"stackit/internal/pkg/examples"
+	"stackit/internal/pkg/flags"
+	"stackit/internal/pkg/globalflags"
+	"stackit/internal/pkg/projectname"
+	"stackit/internal/pkg/services/opensearch/client"
+	opensearchUtils "stackit/internal/pkg/services/opensearch/utils"
+	"stackit/internal/pkg/spinner"
+	"stackit/internal/pkg/utils"
+
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-sdk-go/services/opensearch"
+	"github.com/stackitcloud/stackit-sdk-go/services/opensearch/wait"
+)
+
+const (
+	instanceNameFlag         = "name"
+	enableMonitoringFlag     = "enable-monitoring"
+	graphiteFlag             = "graphite"
+	metricsFrequencyFlag     = "metrics-frequency"
+	metricsPrefixFlag        = "metrics-prefix"
+	monitoringInstanceIdFlag = "monitoring-instance-id"
+	pluginFlag               = "plugin"
+	sgwAclFlag               = "acl"
+	syslogFlag               = "syslog"
+	planIdFlag               = "plan-id"
+	planNameFlag             = "plan-name"
+	versionFlag              = "version"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	PlanName string
+	Version  string
+
+	InstanceName         *string
+	EnableMonitoring     *bool
+	Graphite             *string
+	MetricsFrequency     *int64
+	MetricsPrefix        *string
+	MonitoringInstanceId *string
+	Plugin               *[]string
+	SgwAcl               *[]string
+	Syslog               *[]string
+	PlanId               *string
+}
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an OpenSearch instance",
+		Long:  "Create an OpenSearch instance.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Create an OpenSearch instance with name "my-instance" and specify plan by name and version`,
+				"$ stackit opensearch instance create --name my-instance --plan-name stackit-qa-opensearch-1.2.10-replica --version 2"),
+			examples.NewExample(
+				`Create an OpenSearch instance with name "my-instance" and specify plan by ID`,
+				"$ stackit opensearch instance create --name my-instance --plan-id xxx"),
+			examples.NewExample(
+				`Create an OpenSearch instance with name "my-instance" and specify IP range which is allowed to access it`,
+				"$ stackit opensearch instance create --name my-instance --plan-id xxx --acl 192.168.1.0/24"),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			// Service name and operation needed for error handling
+			service := strings.Split(cmd.Parent().Parent().Use, " ")[0]
```

Comment:
Isn't this just "opensearch"

Reply:
Also below, `operation` is just `"create"`

I guess you're doing this so it can be copy-pasted for other DSA services. I don't like it, because you're making the code harder to read, while not really saving on maintenance effort (since changes here would have to be copy-pasted for other services). It's a middle ground that ends up taking the worst of both worlds. Also, this assumes this command has a parent's parent, and that the grandparent's `Use` starts with the name of the service, which is a very obscure dependency to have.

If you want to take advantage of the DSA APIs being the same, I'd suggest making a `services/dsa` package that holds all the commands core logic. This `RunE` would then just calls `CreateRunE` from `pkg/services/dsa` (or something similar). And that `CreateRunE` would get as input a `dsaClient` interface that opensearch/mariadb/... clients satisfy (which works because the APIs are the same).
Alternatively, you can just not rely on that similarity and have this as a normal command, use `"opensearch"` instead of the `service` variable and so on.

Reply:
We also had it for Postgres and MongoDB Flex but I don't remember the reason, I've replaced it by the explicit service name

In the end we also have it in other places such as the examples, and it can be easily replaced by using the replace tool

Reply:
I'm not sure about having the common code for the DSA, as there's nothing enforcing them to behave similarly AFAIK, so I would suggest keeping it as is and re-evaluate in the future if we have too much effort to maintain them